fdbserver/TLogServer.actor.cpp

```diff
@@ -132,13 +132,18 @@ struct TLogQueue : public IClosable {
 	IDiskQueue* queue;
 	UID dbgid;
 
-	void updateVersionSizes( const TLogQueueEntry& result, TLogData* tLog );
+	void updateVersionSizes( const TLogQueueEntry& result, TLogData* tLog, IDiskQueue::location start, IDiskQueue::location end );
 
 	ACTOR static Future<TLogQueueEntry> readNext( TLogQueue* self, TLogData* tLog ) {
 		state TLogQueueEntry result;
 		state int zeroFillSize = 0;
 
+		// FIXME: initializeRecovery should probably be called on its own by the caller of readNext.
+		bool recoveryFinished = wait( self->queue->initializeRecovery() );
```

Comment:
IKeyValueStore has an init() function which must be called before it is used. We could move disk queue recovery to that function

Reply:
It turns out that hooking this into KeyValueStoreMemory isn't trivial.  Recovery is spawned by the constructor, and init() runs after that.  It seems the most proper to move the constructor logic to init(), and then modify init() to not spawn the recovery actor if diskqueue->initializeRecovery() returns true.  However, this is more refactoring and potential debugging work than seems intended, so I'm going to go leave a FIXME in the constructor that it should be moved to init.